feat: add sink to flashinfer decode

djmmoss opened the pull request:
This PR adds sink support to standard flashinfer decode

```bash
pytest tests/attention/test_decode_sink_attention.py -xs
============================================================================================================================= test session starts ==============================================================================================================================
platform linux -- Python 3.10.12, pytest-9.0.1, pluggy-1.6.0
rootdir: /home/scratch.dmoss_gpu_1/repos/flashinfer
configfile: pytest.ini
collected 110 items

tests/attention/test_decode_sink_attention.py ..............................................................................................................

============================================================================================================================= 110 passed in 8.82s ==============================================================================================================================
```

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added optional auxiliary "sink" (maybe_s_aux/sinks) tensor support across prefill and decode flows, integrating sink values into attention softmax and threading auxiliary buffers through single- and batch-decode (including paged KV) paths.
  * Added input normalization helper to produce a kernel-friendly sink buffer format.

* **Tests**
  * Added comprehensive tests validating sink-attention decoding across batch/single flows, layouts, GQA scenarios, paged KV, and numerical tolerances.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

djmmoss committed:
feat: add sink to flashinfer decode

Co-authored-by: Nitin Gupta <ngupta@groq.com>
Co-authored-by: Duncan Moss <djmmoss@gmail.com>

djmmoss committed:
pre-commit

yzh119 commented:
Note: Adding attention sink support to the CUDA cores template (in `decode.cuh`) is a welcome addition for coverage. However, this template is scheduled for cleanup. Decode attention will default to use_tensor_cores=True, which uses the tensor core-based template defined in prefill.cuh.

djmmoss committed:
Add attention sink support to tensor core template for decode attention

Add maybe_s_aux support to prefill template used for decode attention
when use_tensor_cores=True. Includes updates to params structures,
variant handling, JIT generation, Python wrappers, and comprehensive
test coverage with validation.

djmmoss committed:
Add maybe_s_aux to single decode module additional_tensor_names

Update single decode module's additional_tensor_names to include 'maybe_s_aux'
along with the corresponding 'float' entry in additional_tensor_dtypes.
This matches the batch decode definition and enables sink attention support
for single decode operations.
